### PR TITLE
Introduce run's --docker-no-pull argument

### PR DESCRIPTION
**Pull request by edigaryev**
This brings back the old behavior where we always attempted to pull the latest container image before starting, but lets the user turn it off in case they want to run CI tasks with locally-built container images.

Always pulling images also avoids some tricky-to-diagnose cases like in cirruslabs/cirrus-ci-annotations#22, where running two similar configurations resulted in different results due to stale state in the Docker daemon.

### CODE REVIEW

File: internal/commands/run.go [hunk 1]
```diff
@@ -189,11 +201,15 @@ func newRunCmd() *cobra.Command {
...
 
+	// General flags
 	cmd.PersistentFlags().BoolVar(&dirty, "dirty", false, "if set the project directory will be mounted"+
 		"in read-write mode, otherwise the project directory files are copied, taking .gitignore into account")
 	cmd.PersistentFlags().StringArrayVarP(&environment, "environment", "e", []string{},
 		"set (-e A=B) or pass-through (-e A) an environment variable")
 	cmd.PersistentFlags().BoolVarP(&verbose, "verbose", "v", false, "")
 
+	// Docker-related flags
+	cmd.PersistentFlags().BoolVar(&dockerNoPull, "docker-no-pull", false, "don't pull the images when starting containers")
```
Comment (fkorotkov) [inline internal/commands/run.go:212]:
```suggestion
	cmd.PersistentFlags().BoolVar(&dockerNoPull, "docker-no-pull", false, "don't attempt to pull the images before starting containers")
```

File: internal/executor/options/docker.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package options
+
+type DockerOptions struct {
+	NoPull       bool
+	NoPullImages []string
```
Comment (fkorotkov) [inline internal/executor/options/docker.go:5]: `NoPullImages` sounds a bit weird IMO. What do you think about changing it yo `--docker-skip-pull`?

**Review by fkorotkov — APPROVED** [review]
LGTM. Left a small suggestion about the flag's description. Also left a comment of maybe changing `no` to `skip` but if you think `no` sounds goo than I don't mind merging as is.

File: internal/executor/options/docker.go (hunk 2 above)
Comment (edigaryev) [reply internal/executor/options/docker.go:5]: This is similar to the Ansible's flag: https://docs.ansible.com/ansible/latest/dev_guide/testing_integration.html#non-destructive-tests

I think it is a bit easier to type and I've also seen similarly composed clauses in other Docker commands:

```
% docker run --help | grep '\-\-no'
      --no-healthcheck                 Disable any container-specified HEALTHCHECK
```

```
% docker build --help | grep '\-\-no'
      --no-cache                Do not use cache when building the image
```

**Review by fkorotkov — APPROVED** [review]

**Pull request merged**